uac-api/src/main/java/com/bazaarvoice/emodb/uac/api/AuthUserAccessControl.java

```diff
@@ -0,0 +1,141 @@
+package com.bazaarvoice.emodb.uac.api;
+
+import com.bazaarvoice.emodb.auth.proxy.Credential;
+import com.bazaarvoice.emodb.common.api.UnauthorizedException;
+
+import java.util.Iterator;
+
+/**
+ * Interface for user access control.  The methods in this interface are used for managing users, roles, and permissions.
+ * Note that use of these methods require special permissions from EmoDB.  If a caller invokes a method for which he has
+ * insufficient permissions the method will throw an {@link UnauthorizedException}.  Note that this exception takes
+ * precedence over any other exceptions or responses.  For example, assume a caller does not have permission to view
+ * any roles.  When that caller makes a call to {@link #getRole(String, EmoRoleKey)} he will always get an
+ * UnauthorizedException, regardless of whether the role he's getting actually exists or not.
+ *
+ * The first parameter to every call is the API key of the caller.  It is used for authenticating the caller and
+ * verifying his credentials.  Methods which manage API keys operate on the API key's ID, unless otherwise noted,
+ * and those parameters always appear after the caller's API key.
+ */
+public interface AuthUserAccessControl {
+
+    /**
+     * Gets all roles for which the user has read access.  This method will not throw an UnauthorizedException if
+     * a valid API key is used but does not have permission to view any roles.  In that case it will return
```

Comment:
What is the reasoning behind not throwing a 403 ? Is it a security measure by design?